### PR TITLE
Replace raw queries with query builder design

### PR DESCRIPTION
**Pull request by btry**
modernize some parts of the plugin

### CODE REVIEW

File: inc/entity.class.php [hunk 1]
```diff
@@ -77,26 +77,28 @@ public function post_purgeItem() {
     * @param $ID           integer     entities ID
     * @param $start        integer     first line to retrieve (default 0)
     * @param $limit        integer     max number of line to retrieve (0 for all) (default 0)
-    * @param $sqlfilter    string      to add a SQL filter (default '')
+    * @param $sqlfilter    array       to add a SQL filter (default [])
     * @return array of vouchers
-   **/
-   static function getAllForEntity($ID, $start = 0, $limit = 0, $sqlfilter = '') {
+    */
+   static function getAllForEntity($ID, $start = 0, $limit = 0, $sqlfilter = []): array {
```
Comment (cedric-anne) [inline inc/entity.class.php:83]: `$start` and `$limit` are set to 0 in the only existing call to this method. These parameters could be dropped and related code could be cleaned.

File: inc/ticket.class.php [hunk 2]
```diff
@@ -72,26 +72,28 @@ public static function countForItem(CommonDBTM $item) {
     * @param $ID           integer     tickets ID
     * @param $start        integer     first line to retrieve (default 0)
     * @param $limit        integer     max number of line to retrieve (0 for all) (default 0)
-    * @param $sqlfilter    string      to add a SQL filter (default '')
+    * @param $sqlfilter    array       to add a SQL filter (default [])
     * @return array of vouchers
-   **/
-   static function getAllForTicket($ID, $start = 0, $limit = 0, $sqlfilter = '') {
+    */
+   static function getAllForTicket($ID, $start = 0, $limit = 0, $sqlfilter = []): array {
```
Comment (cedric-anne) [inline inc/ticket.class.php:78]: `$start`, `$limit` and `$sqlfilter` are never set in the only existing call to this method. These parameters could be dropped and related code could be cleaned.

File: inc/ticket.class.php [hunk 3]
```diff
@@ -105,26 +107,28 @@ static function getAllForTicket($ID, $start = 0, $limit = 0, $sqlfilter = '') {
     * @param $ID           integer     plugin_credit_entities_id ID
     * @param $start        integer     first line to retrieve (default 0)
     * @param $limit        integer     max number of line to retrive (0 for all) (default 0)
-    * @param $sqlfilter    string      to add a SQL filter (default '')
+    * @param $sqlfilter    array       to add a SQL filter (default '')
     * @return array of vouchers
    **/
-   static function getAllForCreditEntity($ID, $start = 0, $limit = 0, $sqlfilter = '') {
+   static function getAllForCreditEntity($ID, $start = 0, $limit = 0, $sqlfilter = []): array {
```
Comment (cedric-anne) [inline inc/ticket.class.php:113]: `$start`, `$limit` and `$sqlfilter` are never set in the only existing call to this method. These parameters could be dropped and related code could be cleaned.

**Review by cedric-anne — DISMISSED** [review]

File: inc/ticket.class.php [hunk 4]
```diff
@@ -140,11 +128,16 @@ static function getConsumedForCreditEntity($ID) {
...
-      if ($result = $DB->query($query)) {
+      $request = [
+         'SELECT' => ['SUM' => 'consumed'],
+         'FROM'   => self::getTable(),
+         'WHERE'  => [
+            'plugin_credit_entities_id' => $ID
+         ],
+      ];
+
+      if ($result = $DB->query($request)) {
```
Comment (cedric-anne) [inline inc/ticket.class.php:140]: This will fail. `$DB->query()` expects a string.

Comment (btry) [reply inc/ticket.class.php]: yes, and an other problem fixed with the SUM aggregation

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**